bindings/nodejs/src/functions/uart.cc

```diff
@@ -32,19 +34,63 @@ class SolUART : public JSHandle<SolUART> {
     static const char *jsClassName() { return "SolUART"; }
 };
 
-static void sol_uart_read_callback(void *user_data, struct sol_uart *uart,
-    unsigned char byte_read) {
+static ssize_t sol_uart_on_data_callback(void *user_data,
+    struct sol_uart *uart, const struct sol_buffer *buf) {
     Nan::HandleScope scope;
     sol_uart_data *uart_data = (sol_uart_data *)user_data;
-    Nan::Callback *callback = uart_data->rx_cb;
+    if (!uart_data)
+        return 0;
+
+    Nan::Callback *callback = uart_data->on_data_cb;
     if (!callback)
-        return;
+        return 0;
+
+    struct sol_str_slice slice;
+    slice = sol_buffer_get_slice(buf);
+    Local<Value> buffer =
+        Nan::CopyBuffer(slice.data, slice.len).ToLocalChecked();
```

Comment:
The buffer will have the reference always until it undergoes the garbage collection, so I would say we are copying it on the reference(not on the demand for sure). NewBuffer() is the most efficient way of creating a buffer, but that takes the ownership of the data memory. If I'm allowed to steal the data from the sol_buffer here, I can use the NewBuffer() instead of CopyBuffer() and that would make this much nicer.